### PR TITLE
#153009255 Users should receive a JWT on successful signup

### PR DESCRIPTION
**Pull request by codestaintin**
#### What does this PR do?

Users should receive a JWT on successful signup

#### Description of Task to be completed?

- Install jsonwebtoken package
- Add token secret to .env file
- Create endpoint to test signup functionality
- Create signup function with JWT
- Create middleware for token functionality
- Install chai assertion library and chai-http
- Write test suites for signup functionality

#### How should this be manually tested?
- Clone the repository
- Run `git checkout feature/153009255/jwt-on-signup`
- Run `npm start`
- Send a post request to localhost:3000/api/auth/signup via postman user credentials
- Run `npm test` to run test

#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

- story type: feature
- story id: 153009255
- story title: Users should receive a JWT on successful signup

#### Screenshots (if appropriate)
screen shot 2017-11-29 at 11 16 03 pm
screen shot 2017-11-29 at 11 16 49 pm
screen shot 2017-11-29 at 11 43 19 pm

#### Questions:

N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 [![](https://img.shields.io/badge/Protected_by-Hound-a873d1.svg)](https://houndci.com) [![Coverage Status](https://coveralls.io/repos/github/andela/k2-cfh/badge.svg?branch=chore%2F153009253%2Fintegrate-coveralls-code-coverage-service)](https://coveralls.io/github/andela/k2-cfh?branch=chore%2F153009253%2Fintegrate-coveralls-code-coverage-service)
 
+
```
Comment (damakahindi) [inline README.md]: Why did you add this extra space?

File: app/controllers/users.js [hunk 2]
```diff
@@ -107,54 +108,96 @@ exports.create = function(req, res) {
...
+            message: "Email already exists"
+          });
+        }
+        if (!existingUser) {
+          var user = new User(req.body);
+          user.avatar = avatars[user.avatar];
+          user.save(function (error, newUser) {
+            if (error) {
+              return res.status(400).json({
+                message: "Signup not successful"
```
Comment (damakahindi) [inline app/controllers/users.js]: This should be a server error not a 400 error.

Comment (codestaintin) [reply app/controllers/users.js]: Noted!
